src/layman/util.py

```diff
@@ -504,11 +520,28 @@ def delete_publications(workspace,
                         publ_param,
                         x_forwarded_items=None,
                         actor_name=None,
+                        publications=None,
                         ):
     from layman import authn
+    if (workspace is None or publ_type is None) and not publications:
+        raise ValueError(
+            "Unable call delete_publications without workspace and publ_type, with empty publications")
+    if workspace is not None and publ_type is not None and publications:
+        raise ValueError(
+            "Unable to call delete_publications with both workspace & publ_type set and non-empty publications. "
+            "Use either workspace and publ_type or publications, not both.")
     if not actor_name:
         actor_name = authn.get_authn_username()
-    whole_infos = get_publication_infos(workspace, publ_type, {'actor_name': actor_name, 'access_type': 'write'})
+    if publications is None:
+        whole_infos = get_publication_infos(workspace,
+                                            publ_type,
+                                            {'actor_name': actor_name,
+                                             'access_type': 'write'})
+    else:
+        whole_infos = dict(publications)
+        publication_key = next(iter(publications))[0]
+        workspace = publication_key[0]
+        publ_type = publication_key[1]
     for (_, _, publication) in whole_infos.keys():
```

Comment:
In case of `publications` argument, `publ_type` and `workspace` may also differ per publication. Sou you should read them from `whole_infos` per each publication. Something like
```suggestion
    for (publication_workspace, publication_type, publication_name) in whole_infos:
```